Sources/Nakama/Socket.swift

```diff
@@ -173,6 +201,10 @@ public final class Socket : SocketProtocol {
                             promise.fail(NakamaRealtimeError(error: error))
                         } else if let promise = collatedPromise as? EventLoopPromise<Nakama_Realtime_Status> {
                             promise.fail(NakamaRealtimeError(error: error))
+                        } else if let promise = collatedPromise as? EventLoopPromise<Nakama_Realtime_Party> {
+                            promise.fail(NakamaRealtimeError(error: error))
+                        } else if let promise = collatedPromise as? EventLoopPromise<SwiftProtobuf.Google_Protobuf_Empty> {
+                            promise.fail(NakamaRealtimeError(error: error))
```

Comment:
how come we have a different promise.Fail for every promise type? i.e., why not just once if there is an error?

Reply:
You are right we could just simplify with checking for any error:
```
if let promise = collatedPromise as? EventLoopPromise<Any> {
    promise.fail(NakamaRealtimeError(error: error))
}
```